runtime_builders/builder_util.py

```diff
@@ -57,10 +59,51 @@ def get_file_from_gcs(gcs_file, temp_file):
         return False
 
 
-def write_manifest_file(manifest):
+def verify_and_write_manifest(manifest):
+    """Verify that the provided runtime manifest is valid before publishing.
+
+    Aliases are provided for runtime 'names' that can be included in users'
+    application configuration files: this method ensures that all the aliases
+    can resolve to actual builder files.
+
+    All builders and aliases are turned into nodes in a graph, which is then
+    traversed to be sure that all nodes lead down to a builder node.
+    """
+    try:
+        node_graph = {}
+        for item in manifest.get('runtimes').items():
```

Comment:
This might be a little cleaner with `for key, val in manifest.get('runtimes').iteritems()'